src/platforms/linux/daemon/dbusservice.cpp

```diff
@@ -50,10 +51,17 @@ DBusService::DBusService(QObject* parent) : Daemon(parent) {
 
   // Setup to track user login sessions.
   QDBusConnection bus = QDBusConnection::systemBus();
-  bus.connect("", DBUS_LOGIN_PATH, DBUS_LOGIN_MANAGER, "UserNew", this,
-              SLOT(userCreated(uint, QDBusObjectPath)));
-  bus.connect("", DBUS_LOGIN_PATH, DBUS_LOGIN_MANAGER, "UserRemoved", this,
-              SLOT(userRemoved(uint, QDBusObjectPath)));
+  if (!bus.isConnected()) {
+    logger.error() << "System bus is not connected?";
+  }
+  if (!bus.connect("", DBUS_LOGIN_PATH, DBUS_LOGIN_MANAGER, "UserNew", this,
+                   SLOT(userCreated(uint, QDBusObjectPath)))) {
+    logger.error() << "Failed to connect to UserNew signal";
+  }
+  if (!bus.connect("", DBUS_LOGIN_PATH, DBUS_LOGIN_MANAGER, "UserRemoved", this,
+                   SLOT(userRemoved(uint, QDBusObjectPath)))) {
+    logger.error() << "Failed to connect to UserRemoved signal";
+  }
```

Comment:
Is it possible that more than one condition can occur (i.e. failing to connect to new user and failing to remove said user) within the same execution chain? Wondering if this should be a chain of else-ifs

Reply:
Realistically, none of these error cases should ever occur, this was just added for some extra verbosity while debugging in the event that we have a programming error of some kind. Basically, the system bus should always exist and the only reason for a `connect()` call to fail is because of a bug of some kind...  the bug in question that these logs pointed me towards was QTBUG-108822.

Hypothetically, if the system bus isn't connected then it is almost certain that the following `connect()` calls will fail too.